clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -801,6 +843,43 @@ class SyclKernelFieldChecker
   bool IsInvalid = false;
   DiagnosticsEngine &Diag;
 
+  // Check whether the object is bit-wise copyable
+  bool copyableToKernel(const FieldDecl *FD, const QualType &FieldTy) {
+    // C++ lambda capture already flags non-constant array types.
+    // Here, we check copyability.
+    if (FieldTy->isConstantArrayType()) {
```

Comment:
```suggestion
    if (const auto *CAT = dyn_cast<ConstantArrayType>(FieldTy)) {
```

And so fourth, do the same elsewhere if you're going  to use the type.

Reply:
OK